pkgs/development/libraries/wxwidgets/wxGTK32.nix

```diff
@@ -0,0 +1,170 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchurl
+, gnome2
+, gst_all_1
+, gtk3
+, libGL
+, libGLU
+, libSM
+, libXinerama
+, libXtst
+, libXxf86vm
+, libpng
+, libtiff
+, libjpeg_turbo
+, zlib
+, pcre2
+, pkg-config
+, xorgproto
+, compat28 ? false
+, compat30 ? true
+, unicode ? true
+, withMesa ? lib.elem stdenv.hostPlatform.system lib.platforms.mesaPlatforms
+, withWebKit ? stdenv.isDarwin
+, webkitgtk
+, setfile
+, AGL
+, Carbon
+, Cocoa
+, Kernel
+, QTKit
+, AVFoundation
+, AVKit
+, WebKit
+}:
+let
+  catch = fetchFromGitHub {
+    owner = "wxWidgets";
+    repo = "Catch";
+    rev = "5f5e4cecd1cafc85e109471356dec29e778d2160";
+    hash = "sha256-fB/E17tiAicAkq88Je/YFYohJ6EHJOO54oQaqiR/OzY=";
+  };
+
+  nanosvg = fetchFromGitHub {
+    owner = "wxWidgets";
+    repo = "nanosvg";
+    rev = "ccdb1995134d340a93fb20e3a3d323ccb3838dd0";
+    hash = "sha256-ymziU0NgGqxPOKHwGm0QyEdK/8jL/QYk5UdIQ3Tn8jw=";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "wxwidgets";
+  version = "3.2.1";
+
+  src = fetchFromGitHub {
+    owner = "wxWidgets";
+    repo = "wxWidgets";
+    rev = "v${version}";
+    hash = "sha256-k6td/8pF7ad7+gVm7L0jX79fHKwR7/qrOBpSFggyaI0=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    gst_all_1.gst-plugins-base
+    gst_all_1.gstreamer
+    libpng
+    libtiff
+    libjpeg_turbo
+    zlib
+    pcre2
+  ] ++ lib.optionals stdenv.isLinux [
+    gtk3
+    libSM
+    libXinerama
+    libXtst
+    libXxf86vm
+    xorgproto
+  ]
+  ++ lib.optional withMesa libGLU
+  ++ lib.optional (withWebKit && stdenv.isLinux) webkitgtk
+  ++ lib.optional (withWebKit && stdenv.isDarwin) WebKit
+  ++ lib.optionals stdenv.isDarwin [
+    setfile
+    Carbon
+    Cocoa
+    Kernel
+    QTKit
+    AVFoundation
+    AVKit
+    WebKit
+  ];
+
+  propagatedBuildInputs = lib.optional stdenv.isDarwin AGL;
+
+  configureFlags =
+    [
+      "--disable-precomp-headers"
+      # This is the default option, but be explicit
+      "--disable-monolithic"
+      "--enable-mediactrl"
+      "--with-nanosvg"
+      (
+        if compat28
+        then "--enable-compat28"
+        else "--disable-compat28"
+      )
+      (
+        if compat30
+        then "--enable-compat30"
+        else "--disable-compat30"
+      )
+    ]
+    ++ lib.optional unicode "--enable-unicode"
+    ++ lib.optional withMesa "--with-opengl"
+    ++ lib.optionals stdenv.isDarwin [
+      "--with-osx_cocoa"
+      "--with-libiconv"
+    ]
+    ++ lib.optionals withWebKit [
+      "--enable-webview"
+      "--enable-webviewwebkit"
+    ];
```

Comment:
```suggestion
  configureFlags = [
    "--disable-precomp-headers"
    # This is the default option, but be explicit
    "--disable-monolithic"
    "--enable-mediactrl"
    "--with-nanosvg"
    (if compat28 then "--enable-compat28" else "--disable-compat28")
    (if compat30 then "--enable-compat30" else "--disable-compat30")
  ] ++ lib.optional unicode "--enable-unicode"
  ++ lib.optional withMesa "--with-opengl"
  ++ lib.optionals stdenv.isDarwin [
    "--with-osx_cocoa"
    "--with-libiconv"
  ] ++ lib.optionals withWebKit [
    "--enable-webview"
    "--enable-webviewwebkit"
  ];
```
I initially didn't want to comment this but since darwin is not building I thought I might as well might this perfect.